scripts/o3de/o3de/engine_template.py

```diff
@@ -2045,7 +2045,9 @@ def create_gem(gem_path: pathlib.Path,
         logger.error(f'Gem path {gem_path} already exists.')
         return 1
     else:
-        os.makedirs(gem_path, exist_ok=force)
+        # We can also use an pre-existing folder if it is empty
+        folder_is_empty = gem_path.is_dir() and (len(list(gem_path.iterdir())) == 0)
+        os.makedirs(gem_path, exist_ok=(force or folder_is_empty))
```

Comment:
If we know the folder is empty, then we don't have to call `makedirs` in this case

Reply:
Actually the code above already checks if the directory already exist and is empty on lines 2044-2046.
This change code is not actually needed